server/datastore/mysql/labels.go

```diff
@@ -143,14 +145,28 @@ DELETE FROM label_membership WHERE label_id = ?
 				continue
 			}
 
+			intRegex := regexp.MustCompile(`^[0-9]+$`)
 			// Split hostnames into batches to avoid parameter limit in MySQL.
 			for _, hostIdentifiers := range batchHostnames(s.Hosts) {
+				var stringIdents []string
+				// Start with 0 so id IN (?) always has at least one element.
+				// id = 0 never matches any real host.
+				intIdents := []uint64{0}
+
+				for _, s := range hostIdentifiers {
+					stringIdents = append(stringIdents, s)
+					// Use strconv to check if it's a valid integer
+					if intRegex.MatchString(s) {
+						n, _ := strconv.ParseUint(s, 10, 64)
+						intIdents = append(intIdents, n)
+					}
+				}
+
```

Comment:
In order to force sqlx to send int arguments to the `id in (?)` part of the query, we create a separate list `intIdents` which contains only the identifiers that look like ints.  We seed it with `0` because we can't send an empty list, and `0` won't match anything. Since we'll still be using the index this has negligible performance impact.